pipelines/A10/A10.R

```diff
@@ -0,0 +1,125 @@
+# input: google sheet
+# output: ds$data, ds$dir
+# -------------------------------------------------------------------------
+
+ds <- statbotData::create_dataset("A10")
+ds <- statbotData::download_data(ds)
+
+# -------------------------------------------------------------------------
+# Step: Map Crminal offences to categories
+#   input:  ds$data
+#.  output: dsoffences
+# -------------------------------------------------------------------------
+
+df_offences_and_categories <- ds$data %>%
+  janitor::clean_names(
+  ) %>%
+  dplyr::select(
+    c(straftat)
+  ) %>% dplyr::distinct(
+    straftat
+  )
+
+# Regular expression pattern to match the crime codes
+pattern_offence <- "\\(Art\\. (\\d+)"
+pattern_category_offence <- "^Total"
+
+df_offence_mapping <- df_offences_and_categories %>%
+  dplyr::filter(grepl(pattern_offence, straftat)) %>%
+  dplyr::mutate(kategorie = "")
+
+# Iterate through the crime vector to create the mapping
+offence_mapping <- list()
+categorie <- NULL
+for (offence in rev(ds_offences_and_categories$straftat)) {
+  if (grepl("^Total", offence)) {
+    category <- offence %>% stringr::str_remove("^Total [:digit:]*\\.[:blank:]Titel:[:blank:]")
+  } else if (grepl(pattern_offence, offence)) {
+    offence_mapping[[offence]] <- category
+  }
+}
+
+# Fill the 'titel' column in the tibble based on the mapping
+for (i in seq_along(df_offence_mapping$straftat)) {
+  mapping_key <- df_offence_mapping$straftat[i]
+  if (mapping_key %in% names(offence_mapping)) {
+    df_offence_mapping$kategorie[i] <- offence_mapping[[mapping_key]]
+  }
+}
+df_offence_mapping
+
+# -------------------------------------------------------------------------
+# Step: Clean the data
+#   input: ds$data, df_offence_mapping
+#.  output: ds$cleaned_data
+# -------------------------------------------------------------------------
+
+ds$cleaned_data <- ds$data %>%
+  janitor::clean_names(
+  ) %>%
+  dplyr::filter(
+    !stringr::str_starts(straftat, "Total")
+  ) %>%
+  dplyr::rename(
+    "anzahl" = polizeilich_registrierte_straftaten_gemass_strafgesetzbuch
+  ) %>%
+  dplyr::left_join(df_offence_mapping, by = "straftat") %>%
+  tidyr::pivot_wider(
+    names_from = c("ausfuhrungsgrad", "aufklarungsgrad"),
+    values_from = anzahl,
+    names_prefix = "anzahl_straftaten"
+  ) %>%
+  janitor::clean_names() %>%
+  dplyr::rename(
+    anzahl_straftaten = anzahl_straftaten_ausfuhrungsgrad_total_aufklarungsgrad_total,
+    anzahl_straftaten_unaufgeklaert = anzahl_straftaten_ausfuhrungsgrad_total_unaufgeklart,
+    anzahl_straftaten_aufgeklaert = anzahl_straftaten_ausfuhrungsgrad_total_aufgeklart,
+    anzahl_straftaten_vollendet = anzahl_straftaten_vollendet_aufklarungsgrad_total,
+    anzahl_straftaten_vollendet_unaufgeklart = anzahl_straftaten_vollendet_unaufgeklart,
+    anzahl_straftaten_vollendet_aufgeklart = anzahl_straftaten_vollendet_aufgeklart,
+    anzahl_straftaten_versucht = anzahl_straftaten_versucht_aufklarungsgrad_total,
```

Comment:
I would leave them in for now and let ZHAW decide, since if we remove those then we should remove also the derived columns such as `anzahl_straftaten_versucht_unaufgeklart` etc.